feature/idiomas

HopeAero opened the pull request:
Esta rama contiene feature/migration y hotfix/user-crud

### TESTING

api del backend para hacer las pruebas https://ucab-profile-backend.playoffside.online/api

Creedenciales para hacer las pruebas
`` {
  "email": "davidsalcedo388@gmail.com",
  "password": "123456"
}
``

Ruta de testeo

Primero crear un idioma
![imagen_2023-12-09_214215741](https://github.com/EquestriaReapers/LRT-Backend/assets/73862929/4062a9dd-6a5e-4d1c-a748-908a2e6b8600)

Segundo agregar idiomas o quitar idiomas del perfil logeado
![image](https://github.com/EquestriaReapers/LRT-Backend/assets/73862929/62e41577-e80c-418a-badf-fc3cd2c52a86)

Probar el get paginado para ver si la respuesta te gusto riki ki
![image](https://github.com/EquestriaReapers/LRT-Backend/assets/73862929/bf17c467-598b-47d6-a841-b420f71efca0)

Probar el get por id
![image](https://github.com/EquestriaReapers/LRT-Backend/assets/73862929/204dde4a-a9bb-4312-b9f8-701a35816a28)

LuisRivasW committed:
Metodos de contacto para que revise emmanuel+

HopeAero committed:
Merge branch 'epic/ultimate-v1-merging-branch' into mediosContactos

HopeAero committed:
Method Contact 100% implement

LuisRivasW committed:
Idiomas listo para que lo revise emma

HopeAero committed:
Hotfix users crud

HopeAero committed:
Implement migration

HopeAero committed:
Migration complete

HopeAero committed:
Merge branch 'epic/ultimate-v1-merging-branch' into feature/mediosContactos

HopeAero committed:
Merge branch 'feature/migration' into feature/mediosContactos

HopeAero committed:
Merge branch 'hotfix/users-entity' into feature/mediosContactos

HopeAero committed:
fixes veloz

HopeAero committed:
Merge branch 'epic/ultimate-v1-merging-branch' into idiomas

HopeAero committed:
Remove language-related files and update module
imports

HopeAero committed:
Merge branch 'feature/migration' into idiomas

HopeAero committed:
Merge branch 'hotfix/users-entity' into idiomas

HopeAero committed:
Add document field to RegisterDto and
AuthService.register()

HopeAero committed:
Merge branch 'hotfix/users-entity' into idiomas

HopeAero committed:
Add admin-language tag to SwaggerModule

HopeAero committed:
Update file paths in package.json and
datasource.ts

HopeAero committed:
Merge branch 'feature/migration' into feature/mediosContactos

HopeAero committed:
Merge branch 'feature/migration' into idiomas

HopeAero committed:
Feature language 100% complete

HopeAero committed:
Merge branch 'main' into idiomas

HopeAero committed:
Update .gitignore

HopeAero committed:
Update .gitignore

HopeAero committed:
Revert "Merge branch 'main' into idiomas"

This reverts commit b4e5379e0710ba5f7f7d1386074c95cd0453b74c, reversing
changes made to aabf6c73db2c87a85813967c6a7e1270541a4ff1.

HopeAero committed:
Fix of experience

HopeAero committed:
Toques especiales de skill

HopeAero committed:
Refactor experience deletion method in ExperienceService

HopeAero committed:
Update experience.service.ts

HopeAero committed:
Add DeleteDateColumn to Experience entity and handle NotFoundException in ExperienceService

HopeAero committed:
Update experience removal method to use soft delete

HopeAero committed:
Update file paths in package.json and datasource.ts

HopeAero committed:
Merge branch 'feature/migration' into idiomas

HopeAero committed:
Update package.json scripts

Hector1567XD commented on src/app.module.ts:
```diff
@@ -10,6 +10,8 @@ import { ServeStaticModule } from '@nestjs/serve-static/dist/serve-static.module
 import { ExperienceModule } from './core/experience/experience.module';
 import { SkillsModule } from './core/skills/skills.module';
+import { LanguageModule } from './core/language/language.module';
+import LanguageAction from './core/profiles/service/language.action';
```
Caballero, que es esto de `LenguagueAction` Xd, es `LenguagueService`, los `Services` son los que adentro tienen `Actions` y recordemos los `Action` son de **UN UNICO METODO** no exponen varios métodos publicos por que son para una unica funcion (pueden tener varios métodos privados, eso si).

Hector1567XD commented on src/core/auth/dto/register.dto.ts:
```diff
@@ -13,6 +13,11 @@ export class RegisterDto {
+  @ApiProperty()
+  @IsString()
+  @MinLength(2)
+  document: string;
```
What is document xD?

HopeAero replied:
cedula en ingles

Hector1567XD replied:
La verdad lo veo como un nombre bien poco descriptivo... pero no se como mas almacenarlo, maybe
```suggestion
  documentCi: string;
```
Si no se nos ocurre algo
```suggestion
  cedula: string; //xd
```
o
```suggestion
  citizenDocument: string;
```
Creo que la mejor seria
```suggestion
  documentNumber: string;
```

Hector1567XD replied:
documentNumber me voy mas por esa, pero investiguen a ver que se hace en esos casos

Hector1567XD replied:
o `identification`, idk

Hector1567XD replied:
En ese mismo orden de ideas, debemos hacer que este campo sea obligatorio, que se ponga en el registro y que además sea único, que te evite el registro si detecta a alguien mas con esa misma cédula

Hector1567XD commented on src/core/auth/dto/register.dto.ts:
```diff
@@ -13,6 +13,11 @@ export class RegisterDto {
   @MinLength(1)
   lastname: string;
 
+  @ApiProperty()
```
```suggestion
  @ApiProperty({  description: "Creo que aqui se podian poenr descripciones explicativas de cada vaina para no perderse no xD?" })
```

Hector1567XD commented on src/core/experience/entities/experience.entity.ts:
```diff
@@ -47,4 +48,7 @@ export class Experience {
     default: null,
   })
   endDate: Date;
+
```
```suggestion

@ApiProperty()
```
Maybe?

Hector1567XD replied:
Papa esto no esta resuelto XD

Hector1567XD commented on src/core/experience/experience.controller.ts:
```diff
@@ -108,6 +108,30 @@ export class ExperienceController {
+    description: 'Experience not found',
+  })
+  @ApiInternalServerError()
+  async removeMyExperiencia(
```
```suggestion
  async removeMyExperiencie(
```

Hector1567XD commented on src/core/experience/service/experience.service.ts:
```diff
@@ -72,6 +72,22 @@ export class ExperienceService {
+
+    if (experienceToRemove.affected === 0) {
+      throw new NotFoundException(EXPERIENCE_NOT_FOUND);
+    }
```
Pienso que lo correcto es hacer una pre-validacion e.e
```suggestion
    const experience = await this.experienceRepository.findOne({ id });
    if (!experience) throw new NotFoundException(EXPERIENCE_NOT_FOUND);
    
    try {
        const experienceToRemove = await this.experienceRepository.softDelete({
           profileId: user.id,
           id: id,
        });

        if (experienceToRemove.affected === 0) {
           throw new InternalServerErrorException(UNKOWN_ERROR_DATABASE_MESSAGE);
        }
    } catch (error) {
       if (error instanceof InternalServerErrorException) throw error;
       throw new InternalServerErrorException(UNKOWN_ERROR_DATABASE_MESSAGE);
    }

```

Hector1567XD replied:
Ademas, tambien nos permite hacer una validacion mas especializada
```suggestion
    const experience = await this.experienceRepository.findOne({ id });
    if (!experience) throw new NotFoundException(EXPERIENCE_NOT_FOUND);
    if (experience.profileId !== user.id) throw new ForbiddenException(YOU_NOT_HAVE_PERMISSION_MESSAGE);
    
    try {
        const experienceToRemove = await this.experienceRepository.softDelete({
           profileId: user.id,
           id: id,
        });
        if (experienceToRemove.affected === 0) {
           throw new InternalServerErrorException(UNKOWN_ERROR_DATABASE_MESSAGE);
        }
    } catch (error) {
       if (error instanceof InternalServerErrorException) throw error;
       throw new InternalServerErrorException(UNKOWN_ERROR_DATABASE_MESSAGE);
    }
```
PD: notese que uso `ForbiddenException` en lugar de `UnauthorizedException` ya que Unauthorized se usa cuando el usuario no esta autenticado ni tiene token, Forbidden es para cuando si bien esta autenticado, no tiene acceso a cierto recurso o cierto permiso

Hector1567XD replied:
Me gustaria hicieran esto en este caso en concreto para tener algo de logica de negocio con prevalidaciones guardado por alli si llegamos a necesitarlo para un caso donde si o si necesitemos hacerlo de esta forma

Para el resto de casos donde simplemente se limitan a hacer `experienceToRemove.affected === 0` la verdad no lo se... diria que son deuda tecnica, pero no se si a partir de ahora sea bueno complicarse la vida haciendo esto o irrnos por lo simple (que no estoy seguro funcione en el 100% de los casos)

Queda de parte de la investigacion del manejo de errores en TypeORM que no se quien la esta haciendo

Hector1567XD commented on src/core/language/entities/language.entity.ts:
```diff
@@ -0,0 +1,20 @@
+  @Column()
+  name: string;
+
+  @OneToMany(
+    () => LanguageProfile,
+    (languageProfile) => languageProfile.language,
+  )
+  languageProfile: LanguageProfile[];
```
o.0...

Entiendo que esta es la relación con la PIVOTE de la ManyToMany, es decir la relación entre `Lenguage -> LenguageProfile -> Profile`

Pero... siempre que busquemos un lenguaje se traera el arreglo de TODOS los pivotes que tiene con TODOS los perfiles de la aplicacion? @HopeAero, preocupante

HopeAero replied:
en realidad no lo hace xD si buscas los lenguajes solo te devuelve los lenguajes y ya los que esten registrados en la tabla lenguaje, los de la relacion LenguajeProfile solo es algo que se ve en los get de perfil que es para tener el campo level

HopeAero replied:
Se nota que no usaste el swagger cuando revisaste

Hector1567XD commented on src/core/language/language.controller.spec.ts:
```diff
@@ -0,0 +1,20 @@
+    }).compile();
+
+    controller = module.get<LanguageController>(LanguageController);
+  });
+
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
+  });
```
Esto no basta como prueba unitaria XD, supongo que aun estan por venir las verdaderas pruebas unitarias no?

HopeAero replied:
nop aun no vienen las pruebas pronto si

Hector1567XD commented on src/core/language/language.controller.ts:
```diff
@@ -0,0 +1,94 @@
+    type: [Language],
+  })
+  @ApiInternalServerError()
+  findAll() {
+    return this.languageService.findAll();
+  }
```
Este endpoint!

Deberiamos agregarle un parametro OPCIONAL que sea "name" y que puedas buscar un lenguaje por nombre para hacer el AUTOCOMPLETADO.

Hector1567XD replied:
Esto lo veo super requerido ✅

Hector1567XD replied:
Aunque... tambien se puede hacer un auto-completado en frontend... ya que de todas formas hay una cantidad finita de lenguajes en el mundo no e,e?.... idk, preguntemosle a Luis y a Anna

Hector1567XD replied:
@HopeAero

Hector1567XD replied:
(Hagamoslo en otra tarea)

Hector1567XD replied:
Se hara en https://github.com/EquestriaReapers/LRT-Backend/pull/14

Hector1567XD commented on src/core/language/language.controller.ts:
```diff
@@ -0,0 +1,94 @@
+  remove(@Param('id', ParseIntPipe) id: number) {
+    return this.languageService.remove(id);
+  }
+}
```
```suggestion
```
No se si sea un error de mi parte decir que no hagamos estos endpoints por que si, luego podemos lllegar a necesitarlos, pero...
1. Consumen espacio innecesario en el SWAGGER y en el codigo sabiendo que no los usaremos
2. Insitan a un supuesto admin panel que no tendremos todavia y se pierde tiempo haciendolos cuando realmente tenemos cosas mas importantes que hacer
3. Mejor hacer 3 endpoints que si vamos a usar, ponerle pruebas unitarias, mensajes personalizados, separarlos en distintas clases (actions) hacerlos bellacos a hacer el CRUD completo pero sin pruebas unitarias, sin mensajes personalizados, sin separacion en actions

Hector1567XD replied:
La sugerencia dice que borremoslos, pero ya que estan hechos ps, ok, nos los borremos, pero la verdad no pienso que debemos seguir copiando y pegando endpoints de admin panel de forma mecanica, deberiamos centrarnos en los que realmente vamos a usar y hacerlos COMPLETOS (pruebas unitarias, encapsulamiento, etc)

Hector1567XD commented on src/core/language/language.controller.ts:
```diff
@@ -0,0 +1,94 @@
+  }
+
+  @Get()
+  @Auth(UserRole.GRADUATE || UserRole.ADMIN)
+  @ApiOkResponse({
+    description: 'Returns an array of ALL languages',
+    type: [Language],
+  })
+  @ApiInternalServerError()
+  findAll() {
+    return this.languageService.findAll();
+  }
```
Tambien el problema con esto es que la APP usara un endpoint de ADMIN para poder obtener la lista de lenguajes, si no, por que le dariamos acceso al GRADUATE aqui?... pienso que es un error eso, la APP no deberia poder acceder a ningun endpoint de admin en lo posible

Hector1567XD replied:
⚠️ **Reitero, ya que estan hechos, no los borremos, pero porfavor, tengan eso en cuenta para proximos endpoints, no siempre hace falta completar el CRUD si no se va a usar**

HopeAero replied:
ese endpoint si debo darselo a la APP ellos tienen que registrar en un perfil con un select de los lenguajes existentes

HopeAero replied:
solo le pongo el rol de Admin pq tambien puede usar ese endpoint el admin

Hector1567XD replied:
Si claro, se que va para las apps tambien, pero creeria que deberia estar en la etiqueta /lenguagues/ y no /admin-lenguagues/ pero si, el admin tambien lo necesita, yo haria 2 endpoints distintos aunque hagan lo mismo ps

Pero bueno, son complicaciones quiza innecesarias por ahora

Hector1567XD commented on src/core/language/language.module.ts:
```diff
@@ -0,0 +1,13 @@
+import { Module } from '@nestjs/common';
+import { LanguageService } from './service/language.service';
```
~~Cual seria la diferencia entre el `LenguageService` y el `LenguageAction` e.e?~~

Ya vi la diferencia... el `LenguagueService` creo seria mas bien `LenguagueAdminService` mientras el otro seria `LenguagueService`, pero de hecho, mas correcto me pareceria que el `LenguagueService` se mantenga como `LenguagueService` y el `LenguagueAction` pase a llamarse `LenguagueProfileService`

Pero saben que seria incluso mas arrecho?, aplicar el `Facade Pattern`, se que lo viste en clases @LuisRivasW 7u7

Podriamos hacer un LenguagueServiceFacade, que dentro tendria metodos provenientes de LenguagueService y LenguagueProfileService, pero idk, la verdad no es tan necesario, igual lo ideal es que el LenguagueService sea un Facade en si mismo de los Actions, pero aja....

Hector1567XD replied:
En resumen....
- LenguagueService -> LenguagueService (✅ Hecho)
- LenguagueAction -> LenguagueProfileService (✏️ Pendiente)

Hector1567XD commented on src/core/language/service/language.service.ts:
```diff
@@ -0,0 +1,52 @@
+      ...updateLanguageDto,
+    });
+    if (!language) {
+      throw new NotFoundException(LANGUAGE_NOT_FOUND);
```
Mas bien no?, por que hubo un problema al crearlo
```suggestion
      throw new NotFoundException(ERROR_ON_DATABASE_MESSAGE);
```

Hector1567XD commented on src/core/language/service/language.service.ts:
```diff
@@ -0,0 +1,52 @@
+  async create(createLanguageDto: CreateLanguageDto): Promise<Language> {
+    const language = this.languageRepository.create(createLanguageDto);
+    await this.languageRepository.save(language);
+    return language;
```
Lo he comentado varias veces e.e
```suggestion
    return  await this.languageRepository.save(createLanguageDto);
```

Hector1567XD commented on src/core/profiles/dto/add-language.dto.ts:
```diff
@@ -0,0 +1,12 @@
+
+  @ApiProperty()
+  @IsString()
+  level: string;
```
Deberia ser un ENUM y limitar el nivel a las opciones que le sistema permita, evaluar con @Nannajpg si usaremos A1, A2, B1, B2, etc... o mas bien "Basico" "Intermedio" "Avanzado" "Nativo" ,etc...
```suggestion
  level: string;
```

Hector1567XD replied:
PD: seria bueno hacer que sea nulo.

HopeAero replied:
Bueno esto lo tenemos que preguntar si quieres hacerlo un ENUM pq yo no se si los niveles A1, B1 se usan en todos los idiomas o solo en ingles

Hector1567XD commented on src/core/profiles/entities/language-profile.entity.ts:
```diff
@@ -0,0 +1,35 @@
+
+  @ApiProperty()
+  @Column()
+  level: string;
```
enum e.e

HopeAero replied:
bueno mande una estructura de como sera ese enum de los lvl

Hector1567XD commented on src/core/profiles/profiles.module.ts:
```diff
@@ -5,11 +5,19 @@ import { UsersModule } from '../users/users.module';
+  ],
   controllers: [ProfilesController],
-  providers: [ProfilesService, FindAllPaginateAction],
+  providers: [ProfilesService, FindAllPaginateAction, LanguageAction],
```
Ya hablamos sobre el nuevo nombre :B

Hector1567XD commented on src/core/profiles/service/index.ts:
```diff
@@ -50,12 +52,19 @@ export default class ProfilesService {
+      })),
+    };
+
+    return mappedProfile;
```
Esto se tiene que encapsular... todo el proceso de "Mapeo" de un perfil, en algun "Presenter" o "Adapter". investiguen sobre estos 2 tipos de clase, creo seria un `UserProfilePresenter` y deberia ir en la capa de presentacion incluso

Hector1567XD commented on src/core/profiles/service/language.action.ts:
```diff
@@ -0,0 +1,87 @@
+    languageProfile.profileId = profile.id;
+    languageProfile.level = level;
+
+    return await this.languageProfileRepository.save(languageProfile);
```
Pls
```suggestion
    return await this.languageProfileRepository.save({
       languageId, profileId: profile.id, level
    });
```

Hector1567XD commented on src/core/profiles/service/language.action.ts:
```diff
@@ -0,0 +1,87 @@
+    private readonly profileRepository: Repository<Profile>,
+  ) {}
+
+  async add(addLanguageProfile: AddLanguageDto, user: UserActiveInterface) {
```
Bonita logica de negocios @LuisRivasW, asi como tu UwU ❤️

Hector1567XD replied:
Maybe, para evitar mal entendidos
```suggestion
  async add(addLanguageProfile: AddLanguageProfileDto, user: UserActiveInterface) {
```

Hector1567XD commented on src/core/profiles/service/language.action.ts:
```diff
@@ -0,0 +1,87 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
```
Tengo pendiente de revisar este archivos despues, de momento enviare estos comentarios

Hector1567XD commented on src/core/users/service/users.service.ts:
```diff
@@ -71,6 +71,7 @@ export class UsersService {
       where: { email },
       select: [
         'id',
+        'document',
```
Que es documenttttt 🙈

HopeAero replied:
CEDULAAA manito

Hector1567XD commented on src/core/profiles/service/profiles.queries.ts:
```diff
@@ -26,7 +26,19 @@ export const RANDOM_PROFILES_PAGINATE_QUERY = `
         FROM "profile_skills_skill" "profile_skills"
```
Quedo pendiente de revisar esta madre query

HopeAero replied:
pero si esa madre query ya la revisaste en la hiper epic

Hector1567XD reviewed (changes requested):
Quedo pendiente de hacer pruebas funcionales y revisar 2 archivos, necesitare ayuda para las pruebas funcionales, pls dejen instrucciones de testeo en el PR asi frontends tambien hacen dichas pruebas funcionales

HopeAero committed:
Fixes necesary for mergin

HopeAero commented:
> Quedo pendiente de hacer pruebas funcionales y revisar 2 archivos, necesitare ayuda para las pruebas funcionales, pls dejen instrucciones de testeo en el PR asi frontends tambien hacen dichas pruebas funcionales

ya te cree tu ruta de testeoo homo

HopeAero committed:
Update language messages and DTO name

Hector1567XD commented on src/core/experience/service/experience.service.ts:
```diff
@@ -72,6 +80,30 @@ export class ExperienceService {
+      throw new ForbiddenException(YOU_NOT_HAVE_PERMISSION_MESSAGE);
+
+    try {
+      const experienceToRemove = await this.experienceRepository.softDelete({
+        profileId: user.id,
+        id: id,
+      });
+      if (experienceToRemove.affected === 0) {
+        throw new InternalServerErrorException(INTERNAL_SERVER_ERROR);
```
Bella lógica de negocios ❤.
Un último fix
```suggestion
      const experienceRemoved = await this.experienceRepository.softDelete({
        profileId: user.id,
        id: id,
      });
      if (experienceRemoved.affected === 0) {
        throw new InternalServerErrorException(INTERNAL_SERVER_ERROR);
```
Es una experiencia que ya se elimino, en tiempo pasado, no se va a eliminar

Hector1567XD commented on src/core/profiles/profiles.module.ts:
```diff
@@ -5,11 +5,19 @@ import { UsersModule } from '../users/users.module';
+import { LanguageProfile } from './entities/language-profile.entity';
+import { Language } from '../language/entities/language.entity';
+import { LanguageModule } from '../language/language.module';
+import LanguagueProfileService from './service/languagueProfileService';
```
Creo que el estandar de nombres que seguimos es asi no?
```suggestion
import LanguagueProfileService from './service/languague-profile.service';
```

Hector1567XD commented on src/core/profiles/service/profiles.queries.ts:
```diff
@@ -26,7 +26,19 @@ export const RANDOM_PROFILES_PAGINATE_QUERY = `
         WHERE "profile_skills"."profileId" = "profile"."id"
-      ) AS "skills"
+      ) AS "skills",
+      ARRAY(
+        SELECT DISTINCT ON ("language_profile"."id") json_build_object(
+          'id', "language_profile"."id",
+          'profileId', "language_profile"."profileId",
+          'level', "language_profile"."level",
+          'name', "languages"."name",
+          'languageId', "language_profile"."languageId"
+        )::text
+        FROM "language_profile"
+        JOIN "language" "languages" ON "languages"."id"="language_profile"."languageId"
+        WHERE "language_profile"."profileId" = "profile"."id"
+      ) AS "languageProfile"
```
Servira si hago esto?
```suggestion
      ARRAY(
        SELECT DISTINCT ON (lgp.id) json_build_object(
          id, lgp.id,
          profileId, lgp.profileId,
          level, lgp.level,
          name, lg.name,
          languageId, lgp.languageId
        )::text
        FROM language_profile lgp
        JOIN language lg ON "lg.id=lgp.languageId
        WHERE lgp.profileId = profile.id
      ) AS languageProfile
```

Hector1567XD replied:
Otra cosa, deberia ser languagues unicamente, no languagueProfile ya que si bien lo buscas en la pivote, el frontend sabe que esos son los lenguages del perfil ps... (amenos que le cambies el nombre luego en la capa de presentacion y no lo quieras cambiar aqui)

```suggestion
      ARRAY(
        SELECT DISTINCT ON ("language_profile"."id") json_build_object(
          'id', "language_profile"."id",
          'profileId', "language_profile"."profileId",
          'level', "language_profile"."level",
          'name', "languages"."name",
          'languageId', "language_profile"."languageId"
        )::text
        FROM "language_profile"
        JOIN "language" "languages" ON "languages"."id"="language_profile"."languageId"
        WHERE "language_profile"."profileId" = "profile"."id"
      ) AS languages // <------------------------------
```

HopeAero replied:
esta dudoso que eso funcione con el lgp.id xD, ya si cambiar el AS languages o cambiarlo en la presentacion de la respuesta ta bien

Hector1567XD commented on src/core/profiles/service/languagueProfileService.ts:
```diff
@@ -0,0 +1,89 @@
+
+  async remove(languageId: number, user: UserActiveInterface): Promise<void> {
+    const profile = await this.profileRepository.findOne({
+      relations: ['languages'],
```
Para que seria esto?
```suggestion
```

Hector1567XD commented on src/core/profiles/messages.ts:
```diff
@@ -4,4 +4,12 @@ export const MY_PROFILE_SUCCESFULLY_UPDATED =
+export const PROFILE_SUCCESFULLY_DELETED_LANGUAGE =
+  'Idioma eliminado de tu perfil exitosamente';
 export const ERROR_PROFILE_SKILL_NOT_FOUND = 'Perfil o skill no se encontraron';
+export const ERROR_PROFILE_LANGUAGUE_NOT_FOUND =
+  'Perfil o idioma no se encontraron';
```
¿Por que Perfil o idioma? el perfil tiene otra validacion anterior con otro mensaje si no m equivoco
```suggestion
export const ERROR_LANGUAGUE_NOT_FOUND =
  'El idioma que intentas agregar al perfil no existe';
```

Hector1567XD commented on src/core/profiles/service/languagueProfileService.ts:
```diff
@@ -0,0 +1,89 @@
+      where: { id: languageId },
+    });
+
+    if (!profile || !language) {
+      throw new NotFoundException(ERROR_PROFILE_LANGUAGUE_NOT_FOUND);
+    }
```
ah ya... quiza sea bueno que sean 2 validaciones aparte

Hector1567XD commented:
Nuevamente no hice testing funcional, se lo delego a los frontends en ese aspecto :B

HopeAero committed:
Fix for necesary merge

HopeAero committed:
Document change name

HopeAero committed:
Refactor profile and user DTOs and entities

This commit refactors the profile and user DTOs and entities. It adds a unique constraint to the documentNumber field in both the CreateUserDto and UpdateUserDto classes. Additionally, it adds a unique constraint to the documentNumber column in the User entity.

HopeAero committed:
Add uniqueness validation for documentNumber field in CreateUserDto and UpdateUserDto

HopeAero committed:
Fixes post reu FI

HopeAero committed:
Fix update lastname

HopeAero committed:
Fix typo in lastname property name

HopeAero committed:
Fix typo in ProfilesService lastname field

HopeAero committed:
Fix typo in language profile service

HopeAero committed:
Fix language profile removal in LanguagueProfileService

HopeAero committed:
Remove unused getContactMethods method

HopeAero committed:
Merge branch 'feature/mediosContactos' into idiomas

Yisustxz reviewed (approved):
todo fino caballero 😎

Hector1567XD commented on src/constants/enum/contact/contact.ts:
```diff
@@ -0,0 +1,4 @@
+export enum TypeContact {
+  PHONE = 'phone',
```
Oriana habia quitado telefono? o estoy equivocado?
```suggestion
  WEBSITE = 'website',
```

Hector1567XD commented on src/core/auth/service/auth.service.ts:
```diff
@@ -54,7 +60,9 @@ export class AuthService {
       email,
       password: await bcryptjs.hash(password, 10),
       name,
+      documentNumber,
       lastname,
+      documentNumber,
```
Doble document Number??
